lib/Pinto/Action/Verify.pm

```diff
@@ -14,25 +16,110 @@ use Pinto::Util qw(throw);
 
 #------------------------------------------------------------------------------
 
-extends qw( Pinto::Action );
+extends qw( Pinto::Action::List );
+
+#------------------------------------------------------------------------------
+
+has strict => (
+    is       => 'ro',
+    isa      => Bool,
+    default  => 0,
+);
+
+has files_only => (
+    is       => 'ro',
+    isa      => Bool,
+    default  => 0,
+);
+
 
 #------------------------------------------------------------------------------
 
 sub execute {
     my ($self) = @_;
 
-    my $dist_rs = $self->repo->db->schema->distribution_rs;
+    my $where  = $self->_where;
+    my $attrs  = $self->_attrs;
+    my $method = 'search_' . ( $self->all ? 'package' : 'registration' );
+    my $rs     = $self->repo->db->schema->$method( $where, $attrs );
```

Comment:
Me too. I usually put a special layer in between DBIC and the rest of my application.